### PR TITLE
Fix conditions for red color text in remaining display widgets

### LINKED ISSUE #3012: Coin Machine: Widget text is red when it should not be.
* Time should turn red when less than 10% batch time remains.
* Batch should turn red when <10% batch remains.
* Total should turn red when <10% total remains

Screenshot 2021-12-06 at 23 01 21

### PR DESCRIPTION
**Pull request by chinins**
## Description

This PR fixes the conditions for color coding of the remaining display widgets (time & tokens)

**Changes** 🏗

- simplify props passing in `RemainingTokensValue` 
- update isWarning conditions in `RemainingTokens` & `RemainingTime`

Resolves #3012

### CODE REVIEW

**Review by arrenv — APPROVED** [review]
Working as expected, nice work!

red-updating-less-10-percent

time-colour-10per-red-change


I do have a couple of other things I noticed that you may also be able to include in this PR. However, let me know if you think these should be raised in separate issues.

#1 - Sold out warning on batches is not displaying if the total tokens in the coin machine are sold out.

also-sold-out


#2 - Available tokens on batches not correct.

change-available-on-batch

File: src/modules/dashboard/components/CoinMachine/RemainingDisplayWidgets/RemainingTokens.tsx [hunk 1]
```diff
@@ -65,21 +65,17 @@ const RemainingTokens = ({
...
     }
 
     return <FormattedMessage {...widgetText.placeholder} />;
   }, [widgetText, tokenAmounts]);
 
   const showValueWarning = useMemo(() => {
     if (
-      !isTotalSale ||
-      tokenAmounts?.soldPeriodTokens.gte(tokenAmounts?.maxPeriodTokens)
+      tokenAmounts?.soldPeriodTokens.gt(
```
[the remark this answers is not in the document]
Comment (chinins) [reply src/modules/dashboard/components/CoinMachine/RemainingDisplayWidgets/RemainingTokens.tsx:76]: thank you for pointing that. My linter stopped working at some point and I guess I didn't notice that one after I fixed it.

**Comment by arrenv** [comment]
It is all default, as in the per user limit is 100% and batch size is 40,000.

You do have to let the next batch start though, which is why I skipped forward an hour. I ran the following in terminal.

`curl -H "Content-Type: application/json" -X POST --data '{"jsonrpc":"2.0","method":"evm_increaseTime","params":[3600],"id": 1}' localhost:8545 && curl -H "Content-Type: application/json" -X POST --data '{"jsonrpc":"2.0","method":"evm_mine","params":[]}' localhost:8545`

**Comment by chinins** [comment]
@arrenv, can you please test the token numbers & soldout issues again? Maybe there is something wrong with pulling data from the contracts... The logic is working properly for me (and I had less tokens left than the target):
Screenshot 2021-12-09 at 22 25 52

**Review by ArmandoGraterol — APPROVED** [review]
For me, the color of all 3 widgets is working as expected! And I agree that the second issue should be separated as it's not related to this PR and it _may_ not be a simple, small fix.

**Comment by arrenv** [comment]
@chinins I have been trying to replicate again today with a fresh environment and have not been able to replicate. So, like you said, probably just an environment issue.

Regarding #2 this still seems to be an issue, so I will raise that separately.

**Pull request merged**